core/editable.js

```diff
@@ -1042,12 +1042,16 @@
 								( selected = getSelectedTableList( sel ) ) ) {
```

Comment:
Please, notice that with recent changes - you also changed the value of this expression.

Previously, there is a case, when the returned value was `false`. Currently, the returned value is an object, which makes this expression `truthy`. Because this expression is one of the options in the `OR` statement - it will execute the first block of code, even when there is no `selected.selection`. So it makes a few test cases fail:
- `tests/core/editable/keystrokes/keystrokes`
- `tests/core/editable/keystrokes/delbackspacequirks/collapsed`

To me, the best practice will be to avoid assignments inside `if` statements. It also might be missed by the further readers, that a value changed inside `if` statement.

If you fixed this one - the old tests seem to be fixed, however, tests added in this PR are failing in some cases.

Could you take a look at this problem, analyze the cases and fix it?

Also, could you improve the styling of this `if` statement: brackets in proper line etc.